hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java

```diff
@@ -1100,20 +1100,29 @@ public void rename2(String src, String dst, Options.Rename... options)
   }
 
   @Override // ClientProtocol
-  public boolean truncate(String src, long newLength, String clientName)
-      throws IOException {
+  public boolean truncate(String src, long newLength, String clientName) throws IOException {
     checkNNStartup();
-    stateChangeLog
-        .debug("*DIR* NameNode.truncate: " + src + " to " + newLength);
+    if(stateChangeLog.isDebugEnabled()) {
+      stateChangeLog.debug("*DIR* NameNode.truncate: " + src + " to " +
+          newLength);
+    }
+    CacheEntryWithPayload cacheEntry = RetryCache.waitForCompletion(retryCache, null);
```

Comment:
I think we should have 
``
    namesystem.checkOperation(OperationCategory.WRITE);
``
above the this like other calls and remove this check before lock in FsNamesystem.